### PR TITLE
Switch FormTriggerArea to use CodeMirror query input by default

### PR DESCRIPTION
**Pull request by fkling**
I'm so sorry that I missed that before the release. Unfortunately this
means that this component still uses the Monaco version by default.

I don't think it warrants a patch release since the input is still
usable and CodeMirror can still be enabled site-wide by setting flag
accordingly, but it's still unfortunate `:-/`

## Test plan

CI

## App preview:

- [Web](https://sg-web-fkling-enable-codemirror.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-slbprykjxh.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/enterprise/code-monitoring/components/FormTriggerArea.tsx [hunk 1]
```diff
@@ -127,7 +127,7 @@ export const FormTriggerArea: React.FunctionComponent<React.PropsWithChildren<Tr
 
     const [queryState, setQueryState] = useState<QueryState>({ query: query || '' })
 
-    const editorComponent = useExperimentalFeatures(features => features.editor ?? 'monaco')
+    const editorComponent = useExperimentalFeatures(features => features.editor ?? 'codemirror6')
```
Comment (vovakulikov) [inline client/web/src/enterprise/code-monitoring/components/FormTriggerArea.tsx:130]: @fkling I want to mention that code insights doesn't own code monitoring code base. 

Curios about "active by default" with `?? 'codemirror6'`. Would it be better if we updated [`defaultSettings` object](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/client/web/src/stores/experimentalFeatures.ts?L7%3A16=&utm_product_name=WebStorm&utm_product_version=WebStorm&utm_source=JetBrains-v1.2.2) with `editor: 'codemorror6'`

In this case we don't need to update each consumer of `useExperimentalFeatures(features => features.editor ?? 'monaco')` individually.

**Review by vovakulikov — APPROVED** [review]

File: client/web/src/enterprise/code-monitoring/components/FormTriggerArea.tsx (hunk 1 above)
Comment (fkling) [reply client/web/src/enterprise/code-monitoring/components/FormTriggerArea.tsx:130]: Whoops.... that's what you get when you do late night work and don't even look at the file paths anymore... I have to apologize to @limitedmage then.

> Would it be better if we updated [defaultSettings object](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/client/web/src/stores/experimentalFeatures.ts?L7%3A16=&utm_product_name=WebStorm&utm_product_version=WebStorm&utm_source=JetBrains-v1.2.2) with editor: 'codemorror6'

Right, I have complete forgotten about this 🤦🏻‍♂️ I'll update this too as part of this PR even though likely won't make a difference (release is already cut and in the next release we wanted to disable the flag).

**Review by limitedmage — APPROVED** [review]
LGTM 👍

**Comment by limitedmage** [comment]
I couldn't find a reliable way to test CodeMirror in Jest via user events like typing, since [`jsdom` doesn't support `contenteditable`](https://github.com/jsdom/jsdom/issues/1670), which CodeMirror relies on. I instead switched the tests to use other ways to populate the query input (URL params, props). I think we have sufficient test coverage in integration tests for real user events, so this should be OK.

**Pull request merged**